src/schedule/message_passing.cc

```diff
@@ -477,9 +477,13 @@ std::vector<Expr> MakeBoundCheck(
     CHECK(iv->dom.defined());
     if (!skip_ivar_domain && !iv->dom.same_as(dom)) {
       Expr value = ComputeExpr<Sub>(value_map.at(iv), iv->dom->min);
+      Expr vmin = EvalSet(value, iset_dmap).min();
```

Comment:
Do EvalSet once, instead of multiple times

Reply:
specifically, cache the result of EvalSet and then get min and max

Reply:
done